specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -1798,6 +2708,121 @@
       },
       "x-ms-discriminator-value": "ApiKey"
     },
+    "CognitiveServiceEndpointDeploymentResourceProperties": {
+      "required": [
+        "model"
+      ],
+      "type": "object",
+      "properties": {
+        "model": {
+          "description": "Model used for the endpoint deployment.",
+          "$ref": "#/definitions/EndpointDeploymentModel"
+        },
+        "raiPolicyName": {
+          "type": "string"
+        },
+        "sku": {
+          "$ref": "#/definitions/CognitiveServicesSku"
+        },
+        "versionUpgradeOption": {
+          "type": "string"
+        }
+      }
+    },
+    "CognitiveServicesSku": {
+      "type": "object",
+      "properties": {
+        "capacity": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "family": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "size": {
+          "type": "string"
+        },
+        "tier": {
+          "type": "string"
+        }
+      }
+    },
+    "ComputePolicyDto": {
+      "required": [
+        "name"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ComputePolicyRequest"
+        }
+      ],
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "name": {
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9_]",
+          "type": "string"
+        }
+      }
+    },
+    "ComputePolicyDtoBasicResource": {
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ComputePolicyDto"
+        }
+      }
+    },
+    "ComputePolicyDtoBasicResourceArmPaginatedResult": {
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "type": "string"
+        },
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComputePolicyDtoBasicResource"
+          }
+        }
+      }
+    },
+    "ComputePolicyRequest": {
+      "required": [
+        "azureScope"
+      ],
+      "type": "object",
+      "properties": {
+        "azureScope": {
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9_]",
+          "type": "string"
+        },
+        "definition": {
+          "$ref": "#/definitions/PolicyDefinition"
+        },
+        "effect": {
+          "$ref": "#/definitions/PolicyEffect"
+        },
+        "parameters": {
+          "type": "object"
+        }
```

Comment:
You're using untyped 'object' here. Why?

Reply:
on server-side this is a dictionary of JToken, which I'm not sure how to correctly represent that in swagger, let me ping the engineer who works on that to answer this question

Reply:
OK we will remove those APIs and redesign